contracts/TokenLocking.sol

```diff
@@ -134,21 +134,31 @@ contract TokenLocking is TokenLockingStorage, DSMath { // ignore-swc-123
     emit UserTokenWithdrawn(_token, msg.sender, _amount);
   }
 
-  function punishStakers(address[] memory _stakers, address _beneficiary, uint256 _amount) public onlyReputationMiningCycle {
+  function punishStakers(address[] memory _stakers, uint256 _amount) public onlyReputationMiningCycle {
     address clnyToken = IMetaColony(IColonyNetwork(colonyNetwork).getMetaColony()).getToken();
     uint256 lostStake;
     // Passing an array so that we don't incur the EtherRouter overhead for each staker if we looped over
     // it in ReputationMiningCycle.invalidateHash;
     for (uint256 i = 0; i < _stakers.length; i++) {
       lostStake = min(userLocks[clnyToken][_stakers[i]].balance, _amount);
       userLocks[clnyToken][_stakers[i]].balance = sub(userLocks[clnyToken][_stakers[i]].balance, lostStake);
-      userLocks[clnyToken][_beneficiary].balance = add(userLocks[clnyToken][_beneficiary].balance, lostStake);
       // TODO: Lose rep?
 
-      emit ReputationMinerPenalised(_stakers[i], _beneficiary, lostStake);
+      emit ReputationMinerPenalised(_stakers[i], lostStake);
     }
   }
 
+  function reward(address _recipient, uint256 _amount) public onlyReputationMiningCycle {
+    address clnyToken = IMetaColony(IColonyNetwork(colonyNetwork).getMetaColony()).getToken();
+    // TODO: Gain rep?
```

Comment:
Meh not sure it's necessary -- rep comes from submitting the right hash, we don't need to give rep for defending, that can just be cash.

Reply:
I mean, I'm being paid in the colony's native token for work I'm doing...

Reply:
I guess... I think the difference is that those tokens already paid out rep once (when they were paid out initially), so not sure if a slashed stake (which is what this is) should lead to rep being generated a second time.